tool/tsh/config.go

```diff
@@ -0,0 +1,105 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/gravitational/teleport/api/profile"
+	"github.com/gravitational/teleport/api/utils/keypaths"
+	"github.com/gravitational/trace"
+)
+
+func writeSSHConfig(sb *strings.Builder, clusterName string, knownHostsPath string, proxyHost string, proxyPort string, leaf bool) {
+	// Generate configuration for all Teleport targets.
+	fmt.Fprintf(sb, "# Common flags for all %s hosts\n", clusterName)
```

Comment:
Any reason you didn't use https://golang.org/pkg/text/template/ here?

Reply:
Just a note on this from me. IMO the template package lends itself more to things that are actual templates, i.e they get used multiple times per program run with changing parameters. For one-off string interpolation it's usually slower and less ergonomic.

Reply:
No real reason on my end, so I've just rewritten the function to use a template. It's presumably slower but will only ever be invoked a handful of times so I don't expect it to be very costly. I'm happy to revert if there's strong feelings one way or the other.